docs/concepts/feature-registry.md

```diff
@@ -0,0 +1,117 @@
+
+In case you want to do it in a more customized way, you can use [this Dockerfile](https://github.com/linkedin/feathr/blob/main/FeathrRegistry.Dockerfile) to deploy the REST API and UI. This docker image is more for illustration purpose, and you can customize it further (like building the REST API and UI in separate docker images).
+
+If you use [Azure Resource Provisioning document](../how-to-guides/azure-deployment-arm.md#provision-azure-resources-using-arm-template) to provision the resources, you should be able to access the UI in this website:
+
+```bash
+https://{prefix}webapp.azurewebsites.net
+```
+
+And the corresponding REST API will be:
+
+```bash
+https://{prefix}webapp.azurewebsites.net/api/v1
+```
+
+## Deployment Options
+
+Feathr supports two types of backends for Feature Registry - Azure Purview (Apache Atlas compatible service) and ANSI SQL. Depending on your IT setup, you might choose either of those in the above deployment steps.
+
+Note that if you choose to enable Role-based Access Control (RBAC), you still need to use a SQL service (such as Azure SQL) to store all the RBAC related information.
+
+## Architecture
+
+![Architecture Diagram](../images/architecture.png)
+
+The architecture is as above. More specifically, there are three components for Feathr feature registry:
+
+- Feathr UI, a react based application
+- Feathr REST API, which provides abstraction for different registry providers, as well as role-based access control (RBAC)
+- Different feature registry backends. Currently only Azure Purview and SQL based registry are supported, but more registry providers from the community are welcome.
+
+Both the Feathr UI and the Feathr Python Client interact with the Feathr REST API service. The REST API service then detect if the user has the right access, and route the corresponding request to the registry providers.
+
+## Accessing Registry in Feathr Python Client
+
+In the Feathr python client, if you want to access the registry, you should set the `FEATURE_REGISTRY__API_ENDPOINT` environment variable. The full [document is here](../how-to-guides/feathr-configuration-and-env.md#a-list-of-environment-variables-that-feathr-uses).
+
+Alternatively, you can set the feature registry and the API endpoint in the configuration YAML file:
+
+```yaml
+feature_registry:
+  # The API endpoint of the registry service
+  api_endpoint: "https://feathr-sql-registry.azurewebsites.net/api/v1"
+```
+
+### Register and Listing Features
+
+You can register your features in the centralized registry and share it with other team members who want to consume those features and for further use. You can also use `list_registered_features` to verify if they have been registered successfully.
```

Comment:
Do we have a use case for feature sharing?

Reply:
yes and I make that part explicit. Basically just tell the feature consumers on which features can be consumed.